src/test/java/PagesIvaF/BasePageIvan.java

```diff
@@ -0,0 +1,59 @@
+package PagesIvaF;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import javax.swing.*;
+import java.time.Duration;
+
+public class BasePageIvan {
+
+     WebDriver driver;
+     WebDriverWait wait;
+     Actions actions;
```

Comment:
Make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

_For, example_

```
  protected WebDriver driver;
  protected WebDriverWait wait;
  protected Actions actions;
```